src/components/gcds-checkbox/gcds-checkbox.tsx

```diff
@@ -73,21 +66,130 @@ export class GcdsCheckbox {
       this.errorMessage = "";
     } else if (!this.hasError && this.errorMessage) {
       this.hasError = true;
+    } else if (this.errorMessage == "") {
+      this.hasError = false;
     }
   }
   /**
    * Hint displayed below the label.
    */
   @Prop({ reflect: true, mutable: false }) hint: string;
 
+  /**
+   * Array of validators
+   */
+  @Prop({ mutable: true }) validator: Array<string | ValidatorEntry | Validator<string>>;
+
+  @Watch('validator')
+  validateValidator() {
```

Comment:
Can you turn this into a util function as well, please? It is used by:
- gcds-checkbox
- gcds-textarea
- gcds-input
- gcds-file-uploader
- gcds-select
- gcds-fieldset